### PR TITLE
Save navigation expand/collapse state in the browser's localStorage

### PR DESCRIPTION
**Pull request by skateman**

### CODE REVIEW

File: dist/js/tertiary-nav.js [hunk 1]
```diff
@@ -350,6 +356,20 @@
         });
       },
 
+      loadFromLocalStorage = function () {
+        if (window.localStorage.getItem('patternfly-navigation-primary') == 'collapsed') {
+          toggleNavBarButton.trigger('click');
```
Comment (jeff-phillips-18) [inline dist/js/tertiary-nav.js]: You could just call expandMenu(), or to handle mobile as well, create a function to handle the toggleNavBarButton click  which does what the inline function does now, and then call that function here as well.

File: dist/js/tertiary-nav.js [hunk 2]
```diff
@@ -350,6 +356,20 @@
...
+
+        if (window.localStorage.getItem('patternfly-navigation-secondary') == 'collapsed') {
+          $('.secondary-nav-item-pf.active [data-toggle=collapse-secondary-nav]').trigger('click');
+        }
+
+        if (window.localStorage.getItem('patternfly-navigation-tertiary') == 'collapsed') {
+          $('.tertiary-nav-item-pf.active [data-toggle=collapse-tertiary-nav]').trigger('click');
+        }
+      },
+
```
Comment (jeff-phillips-18) [inline dist/js/tertiary-nav.js]: Again, you could create functions for the click callbacks and call them here.

File: tests-src/_includes/widgets/vertical-nav-with-tertiary.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
   <div class="nav-pf-vertical nav-pf-vertical-with-tertiary-nav">
 {% endif %}
   <ul class="list-group">
-    <li class="list-group-item active">
+    <li class="list-group-item">
```
Comment (jeff-phillips-18) [inline tests-src/_includes/widgets/vertical-nav-with-tertiary.html]: This change is just to remove the initial active item?

Comment (skateman) [inline tests-src/_includes/widgets/vertical-nav-with-tertiary.html]: I had to test somehow the 2nd level collapse with refresh.

**Comment by skateman** [comment]
@jeff-phillips-18 fixed

**Comment by jeff-phillips-18** [comment]
Questions about the use of the $window.localStorage. 
Is this supported for all browsers we support?
Does using a key of 'patternfly-...' cause any application using this to set these values and overwrite each other? If so, should we take a prefix string in the call to setupVerticalTertiaryNavigation?

**Comment by skateman** [comment]
Browser support is available here: http://www.w3schools.com/html/html5_webstorage.asp
The values aren't overridable on a domain-level, but we can make the values parametrizable.

**Comment by jeff-phillips-18** [comment]
First version of IE supported is IE 12. We need to support IE 9 :(

**Comment by skateman** [comment]
Nope, thats MS Edge 12, IE 8 is supported :)

**Comment by jeff-phillips-18** [comment]
My bad...

**Comment by skateman** [comment]
@jeff-phillips-18 so is this mergeable, or should I make the keys parametrizable?

**Comment by jeff-phillips-18** [comment]
Should be fine, we could add that later if need be

**Comment by skateman** [comment]
Thanks! Also could you please rebase your branch with 3.3.3?

**Comment by jeff-phillips-18** [comment]
Do you care if I pull --rebase and force push to keep my commits on top?

**Comment by skateman** [comment]
That's the way i imagined it :)

**Comment by jeff-phillips-18** [comment]
Done!

**Pull request merged**